pkg/cli/cmd/taskrun/describe.go

```diff
@@ -0,0 +1,199 @@
+{{decorate "bold" "Labels"}}:
+{{- range $k, $v := .TaskRun.Labels }}
+ {{ $k }}={{ $v }}
+{{- end }}
+{{- end }}
+{{- if .TaskRun.Annotations }}
+{{decorate "bold" "Annotations"}}:
+{{- range $k, $v := .TaskRun.Annotations }}
+ {{ $k }}={{ $v }}
+{{- end }}
+{{- end }}
+
+📌 {{decorate "underline bold" "Status"}}
+STARTED          DURATION         STATUS
+{{ formatAge .TaskRun.Status.StartTime .Time | printf "% -16s" }} {{ formatDuration .TaskRun.Status.StartTime .TaskRun.Status.CompletionTime | printf "% -16s" }} {{ formatCondition .TaskRun.Status.Conditions }}
+
+⚓ {{decorate "underline bold" "Params"}}
+{{- if ne (len .TaskRun.Spec.Params) 0 }}
+  NAME                          VALUE
+{{- range $i, $p := .TaskRun.Spec.Params }}
+  • {{ $p.Name | printf "% -28s" }}{{ if eq $p.Value.Type "string" }}{{ $p.Value.StringVal }}{{ else if eq $p.Value.Type "array" }}{{ $p.Value.ArrayVal }}{{ else }}{{ $p.Value.ObjectVal }}{{ end }}
+{{- end }}
+{{- end }}
+
+{{- if ne (len .TaskRun.Status.Results) 0 }}
+
+📝 {{decorate "underline bold" "Results"}}
+  NAME                          VALUE
+{{- range $result := .TaskRun.Status.Results }}
+  • {{ $result.Name | printf "% -28s" }}{{ if eq $result.Value.Type "string" }}{{ $result.Value.StringVal }}{{ else if eq $result.Value.Type "array" }}{{ $result.Value.ArrayVal }}{{ else }}{{ $result.Value.ObjectVal }}{{ end }}
+{{- end }}
+{{- end }}
+
+{{- if ne (len .TaskRun.Spec.Workspaces) 0 }}
+
+🗂  {{decorate "underline bold" "Workspaces"}}
+  NAME                SUB PATH            WORKSPACE BINDING
+{{- range $workspace := .TaskRun.Spec.Workspaces }}
+  • {{ $workspace.Name | printf "% -19s" }}{{ if not $workspace.SubPath }}{{ "---" | printf "% -19s" }}{{ else }}{{ $workspace.SubPath | printf "% -19s" }}{{ end }}{{ formatWorkspace $workspace }}
+{{- end }}
+{{- end }}
+`
+
+func describeCommand(p common.Params) *cobra.Command {
+	opts := &options.DescribeOptions{
+		ResourceType: common.ResourceTypeTaskRun,
+	}
+
+	eg := `Describe a TaskRun in namespace 'foo':
+    tkn-results taskrun describe my-taskrun -n foo
+
+Describe a TaskRun in the current namespace
+    tkn-results taskrun describe my-taskrun
+`
+	cmd := &cobra.Command{
+		Use:     "describe [taskrun-name]",
+		Aliases: []string{"desc"},
+		Short:   "Describe a TaskRun",
+		Long:    "Describe a TaskRun by name or UID. If --uid is provided, then TaskRun name is optional.",
+		Annotations: map[string]string{
+			"commandType": "main",
+		},
+		Example: eg,
+		Args: func(_ *cobra.Command, args []string) error {
+			if opts.UID != "" {
+				return nil
+			}
+			if len(args) != 1 {
+				return fmt.Errorf("requires exactly one argument when --uid is not provided")
+			}
+			return nil
+		},
+		PreRunE: func(cmd *cobra.Command, _ []string) error {
+			allNs, _ := cmd.Flags().GetBool("all-namespaces")
+			nsSet := cmd.Flags().Changed("namespace")
+			if allNs && nsSet {
+				return errors.New("cannot use --all-namespaces/-A and --namespace/-n together")
+			}
```

Comment:
If you agree to drop the -A/--all-namespaces option, then that part is redundant as well. Few bits below should be tweaked as well.